src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,777 @@
+
+            # Attribute lists
+            for cluster in read_response.tlvAttributes[endpoint]:
+                # Only verify spec-defined clusters to avoid issues with clusters that may have write-only attributes
+                # Note: Comprehensive verification of wildcard reads is also performed in IDM-10.1 and ACE tests
+                if global_attribute_ids.cluster_id_type(cluster) != global_attribute_ids.ClusterIdType.kStandard:
+                    continue
+
+                returned_attrs = sorted([x for x in read_response.tlvAttributes[endpoint][cluster].keys()])
+                attr_list = sorted([x for x in read_response.tlvAttributes[endpoint][cluster][
+                    ClusterObjects.ALL_CLUSTERS[cluster].Attributes.AttributeList.attribute_id]])
+                asserts.assert_equal(returned_attrs, attr_list,
+                                     f"Mismatch for {cluster} at endpoint {endpoint}")
+
+    def verify_attribute_path(self, read_response: dict, path: AttributePath):
+        """Verify read response for an attribute path.
+
+        Args:
+            read_response: The read response to verify
+            path: The attribute path that was read
+
+        Raises:
+            AssertionError if verification fails
+        """
+        endpoint = path.EndpointId
+        cluster_id = path.ClusterId
+        attribute_id = path.AttributeId
+
+        endpoint_list = [endpoint] if endpoint is not None else list(self.endpoints.keys())
+
+        for ep in endpoint_list:
+            asserts.assert_in(ep, read_response.tlvAttributes,
+                              f"Endpoint {ep} not found in response")
+
+            if cluster_id is not None:
+                asserts.assert_in(cluster_id, read_response.tlvAttributes[ep],
+                                  f"Cluster {cluster_id} not found in endpoint {ep}")
+
+                if attribute_id is not None:
+                    asserts.assert_in(attribute_id, read_response.tlvAttributes[ep][cluster_id],
+                                      f"Attribute {attribute_id} not found in cluster {cluster_id} on endpoint {ep}")
+                else:
+                    # All attributes from the cluster were requested
+                    # Verify AttributeList is present
+                    cluster_obj = ClusterObjects.ALL_CLUSTERS.get(cluster_id)
+                    if cluster_obj and hasattr(cluster_obj.Attributes, 'AttributeList'):
+                        attr_list_id = cluster_obj.Attributes.AttributeList.attribute_id
+                        asserts.assert_in(attr_list_id, read_response.tlvAttributes[ep][cluster_id],
+                                          f"AttributeList not found in cluster {cluster_id} on endpoint {ep}")
+
+                        # Verify that returned attributes match the AttributeList
+                        returned_attrs = sorted([x for x in read_response.tlvAttributes[ep][cluster_id].keys()])

```

Comment:
Should we add an assert here if cluster_id doesn't exist in the endpoint ? Looks like we will raise KeyError instead of an assert ?